### PR TITLE
Added staking events, js query impl & js tests.

### PR DESCRIPTION
**Pull request by CharlieGo88**

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -29,7 +29,8 @@ task("deploy", "Deploy a hedera contract")
       contractName: args.contract,
       // Optional, injected into the constructor, in this case for the "HelloWorld" Contract
       constructorParams: new ContractFunctionParameters()
-        .addAddress(new TokenId(0,0, 34111069).toSolidityAddress())
+        .addAddress(new TokenId(0,0, 34185686).toSolidityAddress())
```
Comment (mattsmithies) [inline hardhat.config.js:32]: fix this or inject the token id through a env option ;-)

File: support/hashgraph.js [hunk 2]
```diff
@@ -137,12 +143,48 @@ const CallContract = async (client, {
   return contractReceipt.status.toString() === 'SUCCESS'
 }
 
+// My bit, it's not as spicey as smithies.
```
Comment (mattsmithies) [inline support/hashgraph.js:146]: Remove comment and add something more descriptive, why using records, why did we decide to seaprate the methods etc

**Review by mattsmithies — CHANGES_REQUESTED** [review]
Looking good so far, fix the bits and we'll merge

**Comment by mattsmithies** [comment]
Ideally your PR description would have some content in it too 😁

File: support/hashgraph.js (hunk 2 above)
Comment (CharlieGo88) [reply support/hashgraph.js]: Added documentation latest commit - insight into functionality, required parameters and what is returned.

File: hardhat.config.js (hunk 1 above)
Comment (CharlieGo88) [reply hardhat.config.js]: Done it chief, updated .env.example to reflect changes.

File: .env.example [hunk 3]
```diff
@@ -3,3 +3,9 @@ HEDERA_PRIVATE_KEY=
 
 ## Example Injected contract ids, post contract deployment
 HTS_CONTRACT_ID=
+STAKABLEPROJECT_CONTRACT_ID=
+HELLOWORLD_CONTRACT_ID=
+
+# Token to be used to interact with treasury & staking.
+# Make sure owner & interacting user has adequate supply.
+STAKABLE_TOKEN_ID =
```
[the remark this answers is not in the document]
Comment (mattsmithies) [reply .env.example:11]: I'll do it myself

**Review by mattsmithies — APPROVED** [review]

**Pull request merged**